apps/aecore/lib/aecore/txs/pool/worker.ex

```diff
@@ -78,4 +115,81 @@ defmodule Aecore.Txs.Pool.Worker do
     {:reply, tx_pool, %{}}
   end
 
+  def add_proof_to_txs(user_txs) do
+    blocks_for_user_txs =
+    for user_tx <- user_txs do
+      get_block_by_txs_hash(user_tx.txs_hash)
+    end
+    merkle_trees =
+    for block <- blocks_for_user_txs do
+      BlockValidation.build_merkle_tree(block.txs)
+    end
+    user_txs_trees = Enum.zip(user_txs, merkle_trees)
+    proof =
+    for user_tx_tree <- user_txs_trees  do
+      {tx, tree} = user_tx_tree
+      key = :erlang.term_to_binary(
+        tx
+        |> Map.delete(:txs_hash)
+        |> Map.delete(:block_hash)
+        |> Map.delete(:block_height)
+        |> Map.delete(:signature)
+        |> TxData.new()
+      )
```

Comment:
this looks weird, can you explain what it does?

Reply:
It removes the keys which are not for the transaction and then turn the map into a TxData stucture.

Reply:
can you refactor `get_txs_for_address` to give you the TxData and not some map, then you don't have to reconstruct your transaction here. If any more attibutes are needed you can return a tuple.